### PR TITLE
Vagrant! You know, for development

### LINKED ISSUE #439: Setup Vagrant for GDI Website Development
**Description**
_We need to improve our ability to get the development site up and running for folks willing to contribute. Using Vagrant or similar tools could improve the startup time while reducing any version/library/OS obstacles that we've originally endured._

**Ask**
_Explore and implement a Vagrant development environment for the GDI site._

**To Do**
- [ ] Research and Identify best option for GDI site
- [ ] Implement
- [ ] Document setup/changes/etc.

**Additional Info**
@amygori is our current go-to for more Vagrant information

### PR DESCRIPTION
**Pull request by amygori**
Needs testing, but I have a working Vagrant box for local development on my machine...see if it works for you!

I think the provisioner has some unnecessary stuff lingering in there, so that may need updates, but I wanted to get this out for some field testing asap.

Closes #439

### CODE REVIEW

**Comment by kstack7** [comment]
Woo! I'm so excited to try this :) Thank you so much, @amygori :D

**Comment by DBNess** [comment]
🙌😍

File: README.md [hunk 1]
```diff
@@ -13,3 +13,19 @@ Girl Develop It is a nonprofit organization that exists to provide affordable an
...
+To  set up the vagrant environment locally:
+1. [Install VirtualBox](https://www.virtualbox.org/)
+2. [install Vagrant](https://www.vagrantup.com/downloads.html)
+3. [clone the git repository](https://github.com/girldevelopit/gdi-website.git)
+4. $ vagrant up
+5. $ vagrant ssh
+6. $ cd /opt/gdi/development
+7. $ bundle install
+8. $ rake db:create db:migrate db:seed
+9. $ bin/rails s
```
Comment (kstack7) [inline README.md:29]: I used the "rails s" command to get it running...so just nix bin (this is more of a note for when move this to actual documentation)

File: README.md [hunk 2]
```diff
@@ -13,3 +13,19 @@ Girl Develop It is a nonprofit organization that exists to provide affordable an
...
 We have migrated our documentation over to Read The Docs! Learn more about how to contribute to our code base by following along here: [http://gdi-new-site.readthedocs.org/](http://gdi-new-site.readthedocs.org/).
+
+###How to Vagrant
+_Note: This will move to the official documentation once the Vagrant dev environment is adopted into the staging branch._
+
+To  set up the vagrant environment locally:
+1. [Install VirtualBox](https://www.virtualbox.org/)
+2. [install Vagrant](https://www.vagrantup.com/downloads.html)
+3. [clone the git repository](https://github.com/girldevelopit/gdi-website.git)
+4. $ vagrant up
```
Comment (kstack7) [inline README.md:24]: We will just want to add that this is a big setup process so to give at least 5 min to see things pull through...

**Comment by kstack7** [comment]
THIS IS EXCITING!! Total success on my end :)

vagrant

**Comment by amygori** [comment]
@kstack7 I am actually floored that it worked perfectly for you!!! SO GREAT! Are you on a PC or Mac? So much agree, the Ruby compilation step should be documented as taking a while. I made that tradeoff to avoid having to use rbenv or rvm to install Ruby. I can update the note in the provisioner to be more than just "a few minutes" but something more visible as a heads up in the documentation would also be good.

**Comment by kstack7** [comment]
@amygori yup! It was very smooth...the load time is totally worth it! I did this on my project-free PC so no chance of conflicts or existing library fun. I can try on my other dev PC and my Mac to help :) So far, so great!

**Comment by amygori** [comment]
EXCELLENT. I will let out a big sigh of relief once @Roenok can get it up and running.

**Comment by imuchnik** [comment]
OMG OMG OMG ! I am going to try it and let you know. This is awesome!

**Comment by Roenok** [comment]
I'll try tonight! I've got some nice, clean virtual machines I can use to test in different environments

**Comment by kstack7** [comment]
Two for two! This time on a Mac :) Lovin' it, @amygori 👍 

screen shot 2016-06-16 at 7 48 49 pm

**Comment by Roenok** [comment]
Yeah, all attempts to install and run failed catastrophically for me. I'm on a Windows 10 laptop. Errors included:
```Timed out while waiting for the machine to boot. This means that
Vagrant was unable to communicate with the guest machine within
the configured ("config.vm.boot_timeout" value) time period.

If you look above, you should be able to see the error(s) that
Vagrant had when attempting to connect to the machine. These errors
are usually good hints as to what may be wrong.

If you're using a custom box, make sure that networking is properly
working and you're able to connect to the machine. It is a common
problem that networking isn't setup properly in these boxes.
Verify that authentication configurations are also setup properly,
as well.

If the box appears to be booting properly, you may want to increase
the timeout ("config.vm.boot_timeout") value.```

```An error occurred in the underlying SSH library that Vagrant uses.
The error message is shown below. In many cases, errors from this
library are caused by ssh-agent issues. Try disabling your SSH
agent or removing some keys and try again.

If the problem persists, please report a bug to the net-ssh project.

timeout during server version negotiating```

```The following SSH command responded with a non-zero exit status.
Vagrant assumes that this means the command failed!

chown `id -u vagrant`:`id -g vagrant` /opt/gdi/development

Stdout from the command:

Stderr from the command:

stdin: is not a tty
chown: changing ownership of ‘/opt/gdi/development’: Not a directory```

**Comment by Roenok** [comment]
Oh man, exciting developments in error messages up in here! Here is what I have discovered:
1. Your machine must support hardware virtualization. For some computers, including Lenovo Thinkpads, this means you must enable this at a BIOS level
2. You must have Git installed. In Windows, using Git within Powershell is not sufficient, as Vagrant cannot run properly within Powershell. Git Bash appears to work
3. Vagrant up will take a very long time to run.

**Comment by Roenok** [comment]
1. Bonus note: MUST use the newest version of VirtualBox (at least above 5.0.12)

That seems to have been the secret sauce

File: motd.sh [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+printf "
+                      (                                       (
+ (               (    )\ )                    (               )\ )    )
+ )\ )   (   (    )\  (()/(     (    )      (  )\             (()/( ( /(
+(()/(   )\  )(  ((_)  /(_))   ))\  /((    ))\((_) (     )    /(_)))\())
+ /(_))_((_)(()\  _   (_))_   /((_)(_))\  /((_)_   )\  /(/(   (_)) (_))/
+(_)) __|(_) ((_)| |   |   \ (_))  _)((_)(_)) | | ((_)((_)_\  |_ _|| |_
+  | (_ || || '_|| |   | |) |/ -_) \ V / / -_)| |/ _ \| '_ \)  | | |  _|
+   \___||_||_|  |_|   |___/ \___|  \_/  \___||_|\___/| .__/  |___| \__|
+                                                     |_|
```
Comment (kstack7) [inline motd.sh:12]: hehehe <3

Comment (amygori) [inline motd.sh:12]: that cracks me up too

**Pull request merged**